xdsl/ir/context.py

```diff
@@ -0,0 +1,533 @@
+from collections.abc import Callable, Iterable
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from xdsl.ir import Attribute, Dialect, Operation
+
+
+def get_all_dialects() -> "dict[str, Callable[[], Dialect]]":
```

Comment:
relatively minor, but if we're splitting things, I'd rather move this to its own file, and it feels a bit more sense to put it in the dialects folder, but that may be a conversation for a different PR

Reply:
Why not, I do think this PR could be split, it's a bit drowning the actual added functionality.
I'll split it after the bases are merged for sanity though :slightly_smiling_face:

Reply:
Yeah, I would agree moving it to its own file, for me it's just because I really don't think it goes well with `MLContext`, which is supposedly a dialect-agnostic thing

Reply:
moved `get_all_dialects` to `xdsl.dialects`, and `MLContext` to root `xdsl`, WDYT?

Reply:
All moved out to:
- #2749